x-pack/plugins/ml/public/plugin.ts

```diff
@@ -99,6 +100,8 @@ export class MlPlugin implements Plugin<MlPluginSetup, MlPluginStart> {
 
   private locator: undefined | MlLocator;
 
+  private navMenuEnabled = true;
```

Comment:
This value has to be an Observable. At the moment there is no way the app will get an update if someone calls `setNavMenuEnabled.

Reply:
This value is set on kibana setup and does not change during the running on kibana as kibana will not switch from being in serverless to not being in serverless.

Reply:
Updated in https://github.com/elastic/kibana/pull/159433/commits/8f4e744df95954be50b562cbb3fff61a386d46d5